packages/loot-core/src/server/importers/ynab5.ts

```diff
@@ -162,6 +162,129 @@ async function importTransactions(
     entityIdMap.set(transaction.id, uuidv4());
   }
 
+  // Compute link between subtransaction transfers and orphaned transaction
+  // transfers. The goal is to match each transfer subtransaction to the related
+  // transfer transaction according to the accounts, date, amount and memo.
+  const orphanTransferMap = data.transactions
+    .filter(
+      transaction =>
+        transaction.transfer_account_id && !transaction.transfer_transaction_id,
+    )
+    .reduce((map, transaction) => {
+      const key =
+        transaction.account_id + '#' + transaction.transfer_account_id;
+      if (!map.has(key)) {
+        map.set(key, [transaction]);
+      } else {
+        map.get(key).push(transaction);
+      }
+      return map;
+    }, new Map<string, YNAB5.Transaction[]>());
+
+  const orphanSubtransfer = data.subtransactions.filter(
+    subtransaction => subtransaction.transfer_account_id,
+  );
+  const orphanSubtransferTrxId = orphanSubtransfer.map(
+    subtransaction => subtransaction.transaction_id,
+  );
+  const orphanSubtransferAcctIdByTrxId = data.transactions
+    .filter(transaction => orphanSubtransferTrxId.includes(transaction.id))
+    .map(
+      transaction =>
+        [transaction.id, transaction.account_id] as [string, string],
+    );
+  const orphanSubtransferAcctIdByTrxIdMap = new Map<string, string>(
+    orphanSubtransferAcctIdByTrxId,
+  );
+  const orphanSubtransferDateByTrxId = data.transactions
```

Comment:
Can we reduce the number times we run through this transaction list by returning an object that has date and account Id.

And we are already running through date.transactions and data.subtransactions in the for loops above your conditions here. We could be building the loopup maps we need in those for loops to reduce some loops.

Reply:
Sure, I removed the additional loops on date.transactions and data.subtransactions and merged the code with the initial loops.